website/src/components/HubSpot.tsx

```diff
@@ -1,16 +1,27 @@
+// This is deprecated. Please use the hubSpot hook instead.
+
 import cookies from 'js-cookie'
 
 export const SOURCEGRAPH_ANONYMOUS_ID_KEY = 'sourcegraphAnonymousUid'
 export const SOURCEGRAPH_SOURCE_URL_KEY = 'sourcegraphSourceUrl'
+
 export interface HubSpotForm {
+    region?: string
     portalId: string
     formId: string
     targetId: string
     onFormSubmit?: (obj: { data: { name: string; value: string }[] }) => void
     onFormReady?: ($form: any) => void
 }
 
-export function createHubSpotForm({ portalId, formId, targetId, onFormSubmit, onFormReady }: HubSpotForm): void {
+export function createHubSpotForm({
+    region,
+    portalId,
+    formId,
+    targetId,
+    onFormSubmit,
+    onFormReady,
+}: HubSpotForm): void {
     const script = document.createElement('script')
```

Comment:
@katjuell I think we should [rethink our current strategy](https://github.com/sourcegraph/about-replatform/blob/main/src/pages/contact/product-specialist.tsx#L20) in the re-platform for integrating this because I think adding scripts in the Layout component is too repetitive to do for each page that needs HubSpot/Chili. I know we inject them into the next/head, but I think loading them within the hook would be cleaner. What do you think?

Reply:
So unfortunately, we need a way to load the forms completely client side in the embedded Markdown as well (as an MDX component that gets passed/ parsed), which means I think two different ways of doing HubSpot however we slice it. I still think the simplest way to handle it is to pass the relevant scripts to the `Layout` component. Unfortunately, we are not able to add the script to the `EmbeddedHubSpot` component either: it needs to already be present for the client side form load work.

Reply:
We'll have to revisit this at some point because I think we can make this work for both cases. As an improvement and since this is HubSpot/Chili specific, I think it would be more efficient to pass a boolean to enable/disable HubSpot/Chili instead of passing scripts every time to prevent unnecessary repetition.

Reply:
Unfortunately, using a hook won't work for our Markdown build, since we need to be able to load these forms individually in the Markdown as MDX components, with embedded scripts that can run completely client side and pass muster with our MDX parser. We can move the script append logic to the `Script` component in `EmbeddedHubSpot`, which will move it out of `Layout` at least (a much better approach!). I have a branch with this change that I can push up to the replatform repo. Happy to continue the conversation on that PR.